node/tests/GlideClusterClient.test.ts

```diff
@@ -61,45 +62,80 @@ const TIMEOUT = 50000;
 describe("GlideClusterClient", () => {
     let testsFailed = 0;
     let cluster: ValkeyCluster;
+    let azCluster: ValkeyCluster;
     let client: GlideClusterClient;
+    let azClient: GlideClusterClient;
     beforeAll(async () => {
         const clusterAddresses = global.CLUSTER_ENDPOINTS;
-        // Connect to cluster or create a new one based on the parsed addresses
-        cluster = clusterAddresses
-            ? await ValkeyCluster.initFromExistingCluster(
-                  true,
-                  parseEndpoints(clusterAddresses),
-                  getServerVersion,
-              )
-            : // setting replicaCount to 1 to facilitate tests routed to replicas
-              await ValkeyCluster.createCluster(true, 3, 1, getServerVersion);
-    }, 40000);
+
+        if (clusterAddresses) {
+            // Initialize current cluster from existing addresses
+            cluster = await ValkeyCluster.initFromExistingCluster(
+                true,
+                parseEndpoints(clusterAddresses),
+                getServerVersion,
+            );
+
+            // Initialize cluster from existing addresses for AzAffinity test
+            azCluster = await ValkeyCluster.initFromExistingCluster(
+                true,
+                parseEndpoints(clusterAddresses),
+                getServerVersion,
+            );
+        } else {
+            cluster = await ValkeyCluster.createCluster(
+                true,
+                3,
+                1,
+                getServerVersion,
+            );
+
+            azCluster = await ValkeyCluster.createCluster(
```

Comment:
maybe you can check the version before and to create it only for valkey 8 and above